openedx/core/lib/block_structure/transformers.py

```diff
@@ -105,3 +108,48 @@ def is_collected_outdated(cls, block_structure):
             )
 
         return bool(outdated_transformers)
+
+    def _categorize_by_type(self):
+        """
+        Returns a tuple of
+            1. list of transformers that implement the transform_block_filter
+            method, and
+            2. list of the rest of the transformers
+        """
+        with_block_filters, other = [], []
+        for transformer in self._transformers:
+            if self._is_block_filter_implemented(transformer):
+                with_block_filters.append(transformer)
+            else:
+                other.append(transformer)
+        return with_block_filters, other
+
+    def _is_block_filter_implemented(self, transformer):
+        """
+        Returns whether the given transformer has its own implementation of
+        the transform_block_filter method.
+        """
+        return transformer.transform_block_filter.__func__ != BlockStructureTransformer.transform_block_filter.__func__
+
+    def _transform_with_block_filters(self, transformers, block_structure):
+        """
+        Transforms the given block_structure using the transform_block_filter
+        method from the given transformers.
+        """
+        if not transformers:
+            return
+
+        combined_filters = functools_reduce(
+            lambda t1_filter, t2_filter: lambda block_key: t1_filter(block_key) and t2_filter(block_key),
+            [t.transform_block_filter(self.usage_info, block_structure) for t in transformers],
+            lambda block_key: True
+        )
```

Comment:
This five-line one-liner is hard to read.   Nested lambdas?  Is that list comprehension being used for side effects, or is the created list actually being used?  Where does the first lambda end?  

Please consider separating it out into named functions.